### Eclipse Workspace Patch 1.0
#P org.eclipse.dltk.javascript.core
Index: parser/com/xored/org/mozilla/javascript/Parser.java
===================================================================
RCS file: /cvsroot/technology/org.eclipse.dltk/javascript/plugins/org.eclipse.dltk.javascript.core/parser/com/xored/org/mozilla/javascript/Parser.java,v
retrieving revision 1.6
diff -u -r1.6 Parser.java
--- parser/com/xored/org/mozilla/javascript/Parser.java	2 Feb 2009 17:03:36 -0000	1.6
+++ parser/com/xored/org/mozilla/javascript/Parser.java	29 Jan 2010 16:47:58 -0000
@@ -504,12 +504,21 @@
 					}
 					fnNode.addParam(s, start);
 					decompiler.addName(s);
+					if (matchToken(Token.COLON)) {
+						// skip type info
+						mustMatchToken(Token.NAME, "msg.no.typeinfo");
+					}
+
 				} while (matchToken(Token.COMMA));
 
 				mustMatchToken(Token.RP, "msg.no.paren.after.parms");
 			}
 			decompiler.addToken(Token.RP);
 
+			if (matchToken(Token.COLON)) {
+				mustMatchToken(Token.NAME, "msg.no.typeinfo");
+			}
+
 			mustMatchToken(Token.LC, "msg.no.brace.body");
 			decompiler.addEOL(Token.LC);
 			body = parseFunctionBody();
@@ -1254,6 +1263,11 @@
 			name.setPosition(nameStart);
 			// omitted check for argument hiding
 
+			if (matchToken(Token.COLON)) {
+				// skip type info
+				mustMatchToken(Token.NAME, "msg.no.typeinfo");
+			}
+
 			if (matchToken(Token.ASSIGN)) {
 				decompiler.addToken(Token.ASSIGN);
 
Index: parser/com/xored/org/mozilla/javascript/resources/Messages.properties
===================================================================
RCS file: /cvsroot/technology/org.eclipse.dltk/javascript/plugins/org.eclipse.dltk.javascript.core/parser/com/xored/org/mozilla/javascript/resources/Messages.properties,v
retrieving revision 1.2
diff -u -r1.2 Messages.properties
--- parser/com/xored/org/mozilla/javascript/resources/Messages.properties	20 Sep 2008 23:32:10 -0000	1.2
+++ parser/com/xored/org/mozilla/javascript/resources/Messages.properties	29 Jan 2010 16:47:58 -0000
@@ -287,6 +287,9 @@
 msg.no.brace.body =\
     missing '{' before function body
 
+msg.no.typeinfo =\
+    missing type info
+
 msg.no.brace.after.body =\
     missing } after function body
 
